Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -582,6 +624,87 @@ private extension HTTPClient {
 
 }
 
+// MARK: - Request Retry Logic
+extension HTTPClient {
+    internal func retryRequestIfNeeded(
+        request: HTTPClient.Request,
+        httpURLResponse: HTTPURLResponse?
+    ) {
+        // retryCount is incremented before the retry is executed, so don't stop retries if
+        // retryCount == self.retryOptions.maxNumberOfRetries
+        guard request.retryCount > self.retryOptions.maxNumberOfRetries else { return }
```

Comment:
shouldn't this be the other way around?

Reply:
Great catch, the correct version should be:

```swift
guard request.retryCount <= self.retryOptions.maxNumberOfRetries else { return }
```

Fixed!